relay-server/src/services/spooler/envelope_stack/sqlite.rs

```diff
@@ -85,46 +90,78 @@ impl SQLiteEnvelopeStack {
         self.batches_buffer_size == 0
     }
 
+    /// Blocks when the `spooling_handle` is active and no data is in the in-memory buffer.
+    ///
+    /// The idea behind such check is that we want to prevent the edge case in which the stack
+    /// appears to be empty because a `push` triggered the spooling of envelopes to the empty disk
+    /// table and because the spooling is async, the size check could see the in-memory buffer and
+    /// disk as both empty.
+    async fn try_wait_spooling(&mut self) -> Result<(), SQLiteEnvelopeStackError> {
+        if self.batches_buffer_size > 0 {
+            return Ok(());
+        }
+
+        let Some(spooling_handle) = mem::take(&mut self.spooling_handle) else {
```

Comment:
Why do we `take` the spooling handle here? Shouldn't every caller of this function wait until the unspool is done?

Reply:
Yes, but the whole implementation of this `EnvelopeStack` relies on the assumption that calls are synchronized and mutually exclusive.

I also do no know if you can wait for a join handle from multiple tasks.